python/ray/data/grouped_data.py

```diff
@@ -195,6 +195,7 @@ def map_groups(
             The return type is determined by the return type of ``fn``, and the return
             value is combined from results of all groups.
         """
+        self._dataset._batch_format = batch_format
```

Comment:
can we move setting `Dataset._batch_format` to `Dataset._map_batches_without_batch_size_validation()`, which is also called by this `map_groups()` method? This will allow us to centralize setting `_batch_format` attribute.

Reply:
After implementing the rule for logical plan, we don't use this attribute in here anymore. Thanks!